### PR TITLE
sumbitting week 1 assessments

### PR DESCRIPTION
**Pull request by cesarcuadro**
# Submitting Week 1 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [ ✅] Coding Challenges
- [ ✅] Coding Concepts
- [ ✅] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -31,24 +38,66 @@ const temp2 = 350
...
+    }
+}
+console.log(temperature(temp1));
+console.log(temperature(temp2));
+console.log(temperature(temp3));
+
 // --------------------2) Create the code that will combine the two arrays and return the length using the test variables provided below.
 
 // Describe your process:
+//Create a variable to combine the two arrays into a new array
```
Comment (elysemontano) [inline code-challenges.js:57]: I would really like to see you include any built-in methods you are considering using to achieve that result.

File: code-challenges.js [hunk 2]
```diff
@@ -31,24 +38,66 @@ const temp2 = 350
...
+//Use a built-in method called .length to determine the length of the new array
+//Print the length of the new array
 
 const padres1984WorldSeriesRuns = [2, 5, 2, 2, 4]
 const padres1998WorldSeriesRuns = [6, 3, 5, 3]
 // Expected output: 9
 
+// let newPadresArray = padres1984WorldSeriesRuns.concat(padres1998WorldSeriesRuns);
+// console.log(newPadresArray.length);
```
Comment (elysemontano) [inline code-challenges.js:66]: I really like that you left your original approach before refactoring your code into a function!  👏

File: code-challenges.js [hunk 3]
```diff
@@ -31,24 +38,66 @@ const temp2 = 350
...
+    let reverseCurrentCohort = currentCohortArray.reverse()
+    return reverseCurrentCohort.join("")
+}
+console.log(cohort(currentCohort))
+
 // --------------------4) Create the code that will return the last index of the given value from the array using the test variables provided below.
 
 // Describe your process:
+//Input: given value
+//Process: Use the given value variable to reference the myNumbers array and iterate through until the given value matches what is found in the array
```
Comment (elysemontano) [inline code-challenges.js:98]: I see iterate here in your pseudocode, but you end up using lastIndexOf.  It is ok to change the plan when you start coding, just be sure to add to your pseudocode the change in direction.

File: code-challenges.js [hunk 4]
```diff
@@ -57,12 +106,28 @@ const givenValue1 = 42
...
+}
+console.log(findValue(givenValue1));
+console.log(findValue(givenValue2));
+
 // --------------------5) Create the code that will sort all the numbers in an array from largest to smallest. Use the test variables provided below.
 
 // Describe your process:
+//Input: use the variables as the argument
+    //each variable has an unsorted array of numbers
+//Process: Normally using .sort() is used for numbers sort in the array from smallest to largest with the argument inside the parentheses being (a, b) => a - b. The values a and b are used to compare two numbers, the comparison continues throughout the array
```
Comment (elysemontano) [inline code-challenges.js:120]: 💯

File: coding-concepts.js [hunk 5]
```diff
@@ -17,38 +17,39 @@ const colors = ["tangerine", "magenta", "lilac", "daffodil"]
...
 
 // --------------------5) What will this log?
 
 const dataTypes = ["number", "string", "Boolean", "undefined"]
 // console.log(typeof dataTypes.length)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: number
+// b) Verify and explain: The .length built-in method will produce the number that represents the amount of data inside an array or string. In this case the amount of data stored in the array is four. Adding on top of number amount of the array, typeof finds out what the primitive data type is. This means that the primitive data type of four is a number.
```
Comment (elysemontano) [inline coding-concepts.js:54]: You crushed this section!!  🤩

File: interview-questions.md [hunk 6]
```diff
@@ -14,44 +14,44 @@ Researched answer:
...
-Researched answer:
+Researched answer: Boolean expressions in JavaScript return true or false. JavaScript operators that will return a boolean value are equality operators, relational operators, and logical operators.
 
 3. What is an index? What is the difference between index and value?
 
-Your answer:
+Your answer: An index is the location of data in either a string or an array. The difference between an index and a value is that an index is the location of the value while the value is what represents the data or information.
 
-Researched answer:
+Researched answer: An index contains the position of a regular expression that matches within a string. A value is the content of each element in an array that must be a JavaScript data type.
```
Comment (elysemontano) [inline interview-questions.md:31]: I am curious to hear more about what you mean by regular expression in this context.

File: interview-questions.md [hunk 7]
```diff
@@ -14,44 +14,44 @@ Researched answer:
...
+Researched answer: An array is a collection of ordered data that can contain any type of information as long as it is a data type that JavaScript recognizes. Strings are a collection of characters inside of quotation marks.
 
-5. Give a brief description of proper pair programming techniques. What are the roles of each person?
+1. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-Your answer:
+Your answer: Pair programming is a method of programming that uses two or more people in order to increase the efficiency and understanding of all parties involved. There are two roles, the driver and navigator. The driver is the one that is typing the code inside of the file while the navigator is guiding and verifying what the driver is doing.
 
-Researched answer:
+Researched answer: The driver uses the keyboard and mouse. The navigator is watching, advising, thinking, is responsible for higher-level input to the code, and catches errors, bugs, and typos.
```
Comment (elysemontano) [inline interview-questions.md:43]: Really great work in this section as well!!  🎉

**Review by elysemontano — APPROVED** [review]
Excellent work Cesar!  I left some comments for you to consider in the future, but overall your work looks fantastic!
Current assessment level is at Competent - You have a really solid understanding of this week's concepts.  To push yourself, consider any possible edge cases that you can cover in your functions and check that the output is correct by making your own test variables.
